go/vt/vtadmin/api.go

```diff
@@ -1050,7 +1065,53 @@ func (api *API) GetSchemaMigrations(ctx context.Context, req *vtadminpb.GetSchem
 	span, ctx := trace.NewSpan(ctx, "API.GetSchemaMigrations")
 	defer span.Finish()
 
-	panic("implement me!")
+	clusterIDs := make([]string, 0, len(req.ClusterRequests))
+	requestsByCluster := make(map[string][]*vtctldatapb.GetSchemaMigrationsRequest, len(req.ClusterRequests))
+
+	for _, r := range req.ClusterRequests {
+		clusterIDs = append(clusterIDs, r.ClusterId)
+		requestsByCluster[r.ClusterId] = append(requestsByCluster[r.ClusterId], r.Request)
+	}
+
+	clusters, _ := api.getClustersForRequest(clusterIDs)
+
+	var (
+		m       sync.Mutex
+		wg      sync.WaitGroup
+		rec     concurrency.AllErrorRecorder
+		results = make([]*vtadminpb.SchemaMigration, 0, len(req.ClusterRequests))
+	)
+
+	for _, c := range clusters {
+		for _, r := range requestsByCluster[c.ID] {
```

Comment:
IMO it's worth a safety check that `requestsByCluster[c.ID]` is not nil — even though it should not be. But we don't want to crash vtadmin if something odd happens, right? Perhaps we're recovering elsewhere though.

Reply:
good call

as to your second question, we do have error recovery in both the http and grpc stacks:
- [grpc](https://github.com/vitessio/vitess/blob/main/go/vt/vtadmin/grpcserver/server.go#L139-L141)
- [http](https://github.com/vitessio/vitess/blob/main/go/vt/vtadmin/api.go#L187-L197)